extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -87,15 +81,26 @@ public Object get(ByteBuffer buf, int position)
   //Note that this is not threadsafe and I don't think it needs to be
   private Union getUnion(ByteBuffer buf, int position)
   {
-    Union union = unions.get(position);
+    Union union = unions.get(buf) != null ? unions.get(buf).get(position) : null;
```

Comment:
In the common case (union already exists) this will unnecessarily call `unions.get(buf)` twice.